### PR TITLE
14_15_16 Directors endpoints added / Tests implemented

### PR DESCRIPTION
**Pull request by Krisssssssssssssssssssssss**

### CODE REVIEW

**Comment by Krisssssssssssssssssssssss** [comment]
#14 
#15 
#16

File: backend/src/main/java/com/example/backend/controller/DirectorController.java [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
+        Director director = directorRequest.toDirector();
+        return DirectorResponse.from(directorService.createDirector(director));
+    }
+    @GetMapping
+    public List<DirectorResponse> getAll(){
+        return directorService.getAllDirectors().stream().map(DirectorResponse::from).collect(Collectors.toList());
+    }
+
+    @GetMapping("/{id}")
+    public DirectorResponse getDirectorByMovieId(@PathVariable @NonNull Long id) {
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/controller/DirectorController.java:32]: The method name is getDirectorByMovieId, but the code returns a Director instance found by its id.

File: backend/src/main/java/com/example/backend/service/DirectorService.java [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+    public Director createDirector(Director director) {
+        return directorRepository.save(director);
+    }
+    public List<Director> getAllDirectors() {
+        return directorRepository.findAll();
+    }
+
+    public Director getDirectorById(Long directorId) {
+        return directorRepository.findById(directorId)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Movie with id " + directorId + " not found"));
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/service/DirectorService.java:27]: Incorrect error message

File: backend/src/main/java/com/example/backend/service/DirectorService.java [hunk 3]
```diff
@@ -1,11 +1,37 @@
...
+    }
+
+    public void deleteDirector(Long directorId) {
+        if (directorRepository.existsById(directorId)) {
+            directorRepository.deleteById(directorId);
+        } else {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Movie with id " + directorId + " does not exist");
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/service/DirectorService.java:34]: Incorrect error message

File: backend/src/main/java/com/example/backend/service/DirectorService.java [hunk 4]
```diff
@@ -1,11 +1,37 @@
 package com.example.backend.service;
 
+import com.example.backend.model.Director;
 import com.example.backend.model.DirectorRepository;
+import com.example.backend.model.Movie;
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/service/DirectorService.java:5]: Unused import

File: backend/src/main/java/com/example/backend/service/DirectorService.java [hunk 5]
```diff
@@ -1,11 +1,37 @@
...
 @Service
 @RequiredArgsConstructor
 public class DirectorService {
     private final DirectorRepository directorRepository;
-}
+
+    public Director createDirector(Director director) {
+        return directorRepository.save(director);
+    }
+    public List<Director> getAllDirectors() {
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/service/DirectorService.java:21]: Class methods should be separated by the empty line.

File: backend/src/main/java/com/example/backend/exception/GlobalExceptionHandler.java [hunk 6]
```diff
@@ -38,6 +41,19 @@ public ErrorResponse handleGlobalException(NoResourceFoundException exception) {
         return new ErrorResponse(exception.getMessage());
     }
 
+    @ExceptionHandler
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ErrorResponse handleValidationException(HandlerMethodValidationException exception) {
+        log.info(exception.getMessage());
+        return new ErrorResponse(exception.getMessage());
+    }
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/exception/GlobalExceptionHandler.java:49]: Class methods should be separated by the empty line.

File: backend/src/main/java/com/example/backend/dto/CreateDirectorRequest.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.example.backend.dto;
+
+import com.example.backend.model.Actor;
+import com.example.backend.model.Director;
+import jakarta.validation.constraints.NotBlank;
+import lombok.NonNull;
+
+public record CreateDirectorRequest(
+
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/dto/CreateDirectorRequest.java:9]: Remove this empty line

File: backend/src/main/java/com/example/backend/exception/GlobalExceptionHandler.java [hunk 8]
```diff
@@ -4,9 +4,12 @@
 import lombok.extern.log4j.Log4j2;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
```
Comment (pavlo-bystrytskyi) [inline backend/src/main/java/com/example/backend/exception/GlobalExceptionHandler.java:7]: Unused import

**Review by pavlo-bystrytskyi — CHANGES_REQUESTED** [review]

**Pull request merged**